src/Tasks/ZipDirectory.cs

```diff
@@ -47,7 +49,7 @@ public override bool Execute()
             {
                 if (destinationFile.Exists)
                 {
-                    if (!Overwrite)
+                    if (!Overwrite || FailIfNotIncremental)
```

Comment:
This feels a little weird to me. If we already zipped the directory, then we should fail as not incremental?

Reply:
Sadly, this task is an not incremental.  The best I could do is error out before File.Delete.

Reply:
I will add a comment.